File: 백준/Bronze/10810. 공 넣기/공 넣기.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int times = sc.nextInt();

        // 바구니 배열 선언
        int[] basArray = new int[n];

        // 바구니 배열 0으로 초기화
        for (int i = 0; i < basArray.length; i++) {
            basArray[i] = 0;
        }

        // 바구니 공 변경
        for (int i = 0; i < times; i++) {
            int first = sc.nextInt();
            int last = sc.nextInt();
            int ballNum = sc.nextInt();

            for (int j = first; j <= last - first + first; j++) {
                basArray[j-1] = ballNum;
            }
        }

        for (int i = 0; i < basArray.length; i++) {
            System.out.print(basArray[i] + " ");
        }
    }
}

File: 백준/Bronze/10810. 공 넣기/공 넣기.cc
#include <iostream>
using namespace std;

int main() {
	int n;
	int times;

	cin >> n >> times;

	int* basket = new int[n];
	
	for (int i = 0; i < n; i++)
	{
		basket[i] = 0;
	}

	int first;
	int last;
	int ballN;

	for (int i = 0; i < times; i++)
	{
		cin >> first >> last >> ballN;

		for (int j = first - 1; j < last; j++)
		{
			basket[j] = ballN;
		}
	}

	for (int i = 0; i < n; i++)
	{
		cout << basket[i] << " ";
	}

	delete[] basket;
}